### DISCUSSION TITLE
Env vars

### DISCUSSION THREAD

**Discussion by advany** [vercel/community#1078] [Help] [answered]
I am trying to access env vars but it keeps saying that process is undefined (inside edge functions)... works with normal functions

**Comment by amyegan** [comment, marked as answer]
Hi @advany! You should be able to access [Environment Variables](https://vercel.com/docs/concepts/functions/edge-functions/edge-functions-api#environment-variables) in [Edge Functions](https://vercel.com/docs/concepts/functions/edge-functions#) with `process.env`.

If you're still having trouble with it, please share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example). I would be happy to take a closer look!

**Comment by advany** [comment]
So I run function below and I get "Unhandled rejection: process is not defined"

export const config = {
    runtime: 'experimental-edge',
};

export default async (request: Request) => {
    return new Response(process.env.SITE,
        {
            status: 200
        });
  };
  
  And vercel.json
  
  {
  "rewrites": [{ "source": "/test.js", "destination": "/api/test.ts" }]
}


I us vercel "vercel": "^27.1.1"

And my .env contains:

SITE="nocodemayo-sf2kjkw58ap1b93462j9"

**Reply by amyegan** [reply]
I created a new edge function in my test app using the code you shared, but I'm not getting the `Unhandled rejection: process is not defined` error. 😞  

I think this is probably caused by a difference in the frameworks and packages we are using. Can you please share an [example repo](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can get a better look at the problem?

**Comment by whopiyush** [comment]
Hey @amyegan , I am running into similar issue, if you are looking for repo to reproduce the `undefined` value, you can check out [officialpiyush/pycz.dev ](https://github.com/officialpiyush/pycz.dev) - all functions are being run on the edge, and process.env.ANY_ENV_VALUE returns undefined in the edge function

**Comment by DanOnCall** [comment]
For anyone who may run into a problem with the `process.env.*` being `undefined` in the edge runtime. 😅 WebStorm was auto-importing the following for us:

```
import * as process from "process";
```

That import then overshadowed the `process` Node.js object. Watch out for those sneaky imports! After we removed that, everything worked as expected.
